common/activity-utilities.js

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable no-prototype-builtins */
+import shuffle from "lodash/shuffle";
+import * as Actions from "~/common/actions";
+
+// NOTE(amine): fetch explore objects
+export const fetchExploreItems = async ({ currentItems, update }) => {
+  const requestObject = {};
+  if (currentItems.length) {
+    if (update) {
+      requestObject.latestTimestamp = currentItems[0].createdAt;
```

Comment:
Things aren't necessarily ordered by date anymore, so this method won't work. In order to get the latest/earliest timestamp you'll need to search through all the activity items

Reply:
Here I'm not using the groups, currentItems are the items from getActivity/getExplore response before the grouping happens. I assume those are ordered by date

Reply:
Ahh ok yep that's fine then